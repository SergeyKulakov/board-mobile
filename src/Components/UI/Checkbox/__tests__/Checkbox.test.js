import React from 'react'
import { shallow } from 'enzyme'
import { expect as chaiExpect } from 'chai'

import 'jest-styled-components'

import { Checkbox as Component } from '../index'

// todo test not runned
describe('Checkbox', () => {
  const defaultProps = {
    checked: false,
    text: 'default text',
    onChange: () => null,
  }

  it('snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should be current render text', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    chaiExpect(
      wrapper
        .find('[testID="textContainer"]')
        .children()
        .text(),
    ).to.be.equal(defaultProps.text)
  })
})
