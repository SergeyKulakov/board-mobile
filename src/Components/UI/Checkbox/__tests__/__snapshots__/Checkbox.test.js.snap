// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox snapshot 1`] = `
<Styled(View)>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    testID="touchableField"
  >
    <Styled(View)
      isActive={false}
    >
      <Icon
        color="#fff"
        name="check"
        size={23}
        visible={false}
      />
    </Styled(View)>
  </TouchableOpacity>
  <Styled(Text)
    testID="textContainer"
  >
    default text
  </Styled(Text)>
</Styled(View)>
`;
